src/toil/lib/docker.py

```diff
@@ -59,8 +59,8 @@ def apiDockerCall(job,
                   entrypoint=None,
                   detach=False,
                   log_config=None,
-                  auto_remove=None,
-                  remove=False,
+                  auto_remove=False,
+                  remove=True,
```

Comment:
Actually, a lot of these kwargs are borrowed from the Docker API, and we sort of refer to it in the docstring as useful documentation for their semantics. If we're now differing more substantially from the default behavior of the Docker API, maybe we need to write something about differences from the normal Docker API in our docstring.